code/modules/reagents/Chemistry-Goon-Medicine.dm

```diff
@@ -0,0 +1,335 @@
+	result_amount = 5
+
+datum/reagent/perfluorodecalin
+	name = "Perfluorodecalin"
+	id = "perfluorodecalin"
+	description = "Heals 25 OXY damage, but you can't talk. 33% chance of healing 1 BRUTE and 1 BURN."
+	reagent_state = LIQUID
+	color = "#C8A5DC" // rgb: 200, 165, 220
+	metabolization_rate = 0.2
+
+datum/reagent/perfluorodecalin/on_mob_life(var/mob/living/carbon/human/M as mob)
+	if(!M) M = holder.my_atom
+	M.adjustOxyLoss(-25*REM)
+	M.silent = max(M.silent, 5)
+	if(prob(33))
+		M.adjustBruteLoss(-1*REM)
+		M.adjustFireLoss(-1*REM)
+	..()
+	return
+
+/datum/chemical_reaction/perfluorodecalin
+	name = "Perfluorodecalin"
+	id = "perfluorodecalin"
+	result = "perfluorodecalin"
+	required_reagents = list("hydrogen" = 1, "fluorine" = 1, "oil" = 1)
+	result_amount = 3
+	required_temp = 370
+	mix_message = "The mixture rapidly turns into a dense pink liquid."
+
+datum/reagent/ephedrine
+	name = "Ephedrine"
+	id = "ephedrine"
+	description = "Stun reduction per cycle, increases run speed slightly, minor stamina regeneration buff, stabilizes crit."
+	reagent_state = LIQUID
+	color = "#C8A5DC" // rgb: 200, 165, 220
+	metabolization_rate = 0.3
+
+datum/reagent/ephedrine/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	M.status_flags |= GOTTAGOFAST
+	if(M.losebreath >= 10)
+		M.losebreath = max(10, M.losebreath-5)
+	M.AdjustParalysis(-1)
+	M.AdjustStunned(-1)
+	M.AdjustWeakened(-1)
+	M.adjustStaminaLoss(-1*REM)
+	..()
+	return
+
+/datum/chemical_reaction/ephedrine
+	name = "Ephedrine"
+	id = "ephedrine"
+	result = "ephedrine"
+	required_reagents = list("sugar" = 1, "oil" = 1, "hydrogen" = 1, "diethylamine" = 1)
+	result_amount = 4
+	mix_message = "The solution fizzes and gives off toxic fumes."
+
+datum/reagent/diphenhydramine
+	name = "Diphenhydramine"
+	id = "diphenhydramine"
+	description = "Stun reduction per cycle, increases run speed slightly, minor stamina regeneration buff, stabilizes crit."
+	reagent_state = LIQUID
+	color = "#C8A5DC" // rgb: 200, 165, 220
+datum/reagent/diphenhydramine/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	M.drowsyness -= 1
+	M.jitteriness -= 1
+	..()
+	return
+
+/datum/chemical_reaction/diphenhydramine
+	name = "Diphenhydramine"
+	id = "diphenhydramine"
+	result = "diphenhydramine"
+	required_reagents = list("oil" = 1, "carbon" = 1, "bromine" = 1, "diethylamine" = 1, "ethanol" = 1)
+	result_amount = 4
+	mix_message = "The mixture dries into a pale blue powder."
+
+datum/reagent/morphine
+	name = "Morphine"
+	id = "morphine"
+	description = "100% chance per cycle of healing 1 point each of OXY and TOX damage."
+	reagent_state = LIQUID
+	color = "#C8A5DC" // rgb: 200, 165, 220
+	var/cycle_count = 0
+
+datum/reagent/morphine/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	M.status_flags |= GOTTAGOFAST
+	if(cycle_count == 36)
+		M.sleeping += 1
+	if(volume > 20)
+		var/obj/item/I = M.get_active_hand()
+		if(I)
+			M.drop_item()
+	cycle_count++
+	..()
```

Comment:
descriptions do not show up in game in any mean so I got lazy with updating them. imo they should be scrapped entirely but that's for a separate PR